plugins/input/netping/netping.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2021 iLogtail Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package netping
+
+import (
+	"fmt"
+	"math"
+	"net"
+	"time"
+
+	"github.com/alibaba/ilogtail"
+	"github.com/alibaba/ilogtail/helper"
+	"github.com/alibaba/ilogtail/pkg/logger"
+
+	goping "github.com/go-ping/ping"
+)
+
+type Result struct {
+	Valid    bool // if the result is meaningful for count
+	Label    string
+	Type     string
+	Total    int
+	Success  int
+	Failed   int
+	MinRTTMs float64
+	MaxRTTMs float64
+	AvgRTTMs float64
+}
+
+type ICMPConfig struct {
+	Src    string `json:"src"`
+	Target string `json:"target"`
+	Count  int    `json:"count"`
+}
+
+type TCPConfig struct {
+	Src    string `json:"src"`
+	Target string `json:"target"`
+	Port   int    `json:"port"`
+	Count  int    `json:"count"`
+}
+
+// NetPing struct implements the MetricInput interface.
+// The plugin has a counter field, which would be increment every 5 seconds.
+// And, the value of this counter will be sent as metrics data.
+type NetPing struct {
+	IcmpPrivileged  bool
+	HasConfig       bool
+	context         ilogtail.Context
+	IntervalSeconds int          `json:"interval_seconds" comment:"the interval of ping/tcping, unit is second,must large than 30"`
+	ICMPConfigs     []ICMPConfig `json:"icmp" comment:"the icmping config list, example:  {\"src\" : \"${IP_ADDR}\",  \"target\" : \"${REMOTE_HOST}\", \"count\" : 3}"`
+	TCPConfigs      []TCPConfig  `json:"tcp" comment:"the tcping config list, example: {\"src\" : \"${IP_ADDR}\",  \"target\" : \"${REMOTE_HOST}\", \"port\" : ${PORT}, \"count\" : 3}"`
+}
+
+const MinIntervalSecods = 30 // min inteval should large than 30s
+
+// Get preferred outbound ip of this machine
+func getOutboudIP() (net.IP, error) {
```

Comment:
How about to use ` util.GetIPAddress()` ？

Reply:
replaced